python/hsfs/core/validation_report_engine.py

```diff
@@ -15,14 +15,20 @@
 #
 from __future__ import annotations
 
-from typing import List, Union
+import importlib.util
+from typing import TYPE_CHECKING, List, Union
 
-import great_expectations as ge
 from hsfs import client, util
 from hsfs.core import validation_report_api
 from hsfs.validation_report import ValidationReport
 
 
+HAS_GREAT_EXPECTATIONS = False
+if importlib.util.find_spec("great_expectations") or TYPE_CHECKING:
+    HAS_GREAT_EXPECTATIONS = True
+    import great_expectations
```

Comment:
This is strange. If we don't have GE, but have TYPE_CHECKING, it will set HAS_GREAT_EXPECTATIONS to True (but then fail on import). It seems that `or TYPE_CHECKING` should be removed.

I propose to add `HAS_GREAT_EXPECTATIONS = importlib.util.find_spec("great_expectations") is not None` to core.constants instead. Also, maybe an overkill, but it can make sense to create a decorator:
```
def uses_great_expectations(f):
    @functools.wraps(f)
    def g(*args, **kwds):
        if not HAS_GREAT_EXPECTATIONS:
            raise ModuleNotFoundError(great_expectations_not_installed_message)
        return f(*args, **kwds)
    return g
```

Reply:
Regarding the "or TYPE_CHECKING", TYPE_CHECKING will be true everywhere else so it will fail in every file we have the check. Why are you concern about it failing here?

Decorator is a good idea, maybe something we should add systematically for "special dependencies". We can start with GE, see how it works out for now.

Reply:
Ok, I have thought more about this and I think it is fine to use `or TYPE_CHECKING` here. I initially thought that it is completely redundant, but it actually changes the error type checker shows (import vs use), which I guess is a bit better.